pkg/pool-manager/pool.go

```diff
@@ -213,6 +221,32 @@ func (p *PoolManager) isInstanceOptedIn(namespaceName, mutatingWebhookConfigName
 	return false, nil
 }
 
+func (p *PoolManager) getOptMode(mutatingWebhookConfigName, webhookName string) (OptMode, error) {
+	mutatingWebhookConfiguration, err := p.kubeClient.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(context.TODO(), mutatingWebhookConfigName, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to get mutatingWebhookConfig %s", mutatingWebhookConfigName)
+	}
+
+	for _, webhook := range mutatingWebhookConfiguration.Webhooks {
+		if webhook.Name == webhookName {
+			for _, expression := range webhook.NamespaceSelector.MatchExpressions {
+				if reflect.DeepEqual(expression, metav1.LabelSelectorRequirement{Key: webhookName, Operator: "In", Values: []string{"allocate"}}) {
+					return OptInMode, nil
+				} else if reflect.DeepEqual(expression, metav1.LabelSelectorRequirement{Key: webhookName, Operator: "NotIn", Values: []string{"ignore"}}) {
+					return OptOutMode, nil
+				}
+			}
+
+			// opt-in can technically also be defined with matchLabels
+			if value, ok := webhook.NamespaceSelector.MatchLabels[webhookName]; ok && value == "allocate" {
+					return OptInMode, nil
+			}
```

Comment:
Are you going to drop one of the method's to check the opt mod?

Reply:
I locally removed the code added in `virtualmachine_controller.go` and re-ran the tests to make sure the test the added tests fail. 
would you like me to do it on the PR so you can see that?